src/controls/Ft3asApp.tsx

```diff
@@ -65,12 +66,92 @@ export default function Ft3asApp() {
     }
     // useEffect(()=>{setChecklistDoc(checklistDoc)}, [checklistDoc]);
 
+    const downloadFile = () => {
+        console.log('Test')
+        const fileName = 'review.json'
+        const fileType = 'text/json'
+        var data = JSON.stringify(checklistDoc) 
+        // Create a blob with the data we want to download as a file
+        const blob = new Blob([data], { type: fileType })
+        // Create an anchor element and dispatch a click event on it
+        // to trigger a download
+        const a = document.createElement('a')
+        a.download = fileName
+        a.href = window.URL.createObjectURL(blob)
+        const clickEvt = new MouseEvent('click', {
+            view: window,
+            bubbles: true,
+            cancelable: true,
+        })
+        a.dispatchEvent(clickEvt)
+        a.remove()
+    }
+
+    const uploadFile = (ev: React.MouseEvent<HTMLElement, MouseEvent> | React.KeyboardEvent<HTMLElement> | undefined) => {
+        ev?.persist();
+
+        Promise.resolve().then(() => {
+            const inputElement = document.createElement('input');
+            inputElement.style.visibility = 'hidden';
+            inputElement.setAttribute('type', 'file');
+
+            document.body.appendChild(inputElement);
+
+            const target = ev?.target as HTMLElement | undefined;
+
+            if (target) {
+                setVirtualParent(inputElement, target);
+            }
+
+            inputElement.click();
+            inputElement.onchange= (e) => {
+                if ((e.target as HTMLInputElement).files === null) {
+                    return
+                }
+                else {
+                    var files = (e.target as HTMLInputElement).files;
+                    var file = files?.item(0);
+
+                    if (file) {
+                    
+                        var reader = new FileReader();
+                        reader.onload = function(event) {
+                            const contents = event?.target?.result
+                            const doc = JSON.parse(contents as string) as IChecklistDocument
+                            setChecklistDoc(doc)
+                        };
+            
+                        (e.target as HTMLInputElement).value = ''
+            
+                        reader.readAsText(file);
+                    } else {
+                        console.error(
+                          'File could not be uploaded. Please try again.'
```

Comment:
Could we give more info on why a file could not be uploaded? 
Also, since this is a console.log, not sure whether this will be nicely visible to the end user that something went wrong.

Reply:
I don't know, may require some research. Meanwhile let's merge the content and will improve if possible